src/otaclient/ota_core.py

```diff
@@ -321,14 +315,20 @@ def _download_file(
                 ),
                 max_idle_timeout=cfg.DOWNLOAD_INACTIVE_TIMEOUT,
             ),
-        ) as _mapper:
+        )
+        try:
             _next_commit_before, _report_batch_cnt = 0, 0
             _merged_payload = UpdateProgressReport(
                 operation=UpdateProgressReport.Type.DOWNLOAD_REMOTE_COPY
             )
-
             for _done_count, _fut in enumerate(
-                _mapper.ensure_tasks(_download_file, download_list), start=1
+                _mapper.ensure_tasks(
+                    self._download_file,
+                    resource_meta.iter_resources(
+                        batch_size=cfg.MAX_CONCURRENT_DOWNLOAD_TASKS
+                    ),
+                ),
+                start=1,
```

Comment:
seems `start=1` is unnecessary.

Reply:
Although it doesn't affect the result, but here we are counting the items we downloaded, so we need to start from 1.